spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -686,61 +688,66 @@ def add_electrical_series(
         """
         if nwbfile is not None:
             assert isinstance(nwbfile, NWBFile), "'nwbfile' should be of type pynwb.NWBFile!"
+
         assert buffer_mb > 10, "'buffer_mb' should be at least 10MB to ensure data can be chunked!"
+
         if not nwbfile.electrodes:
             se.NwbRecordingExtractor.add_electrodes(recording, nwbfile, metadata)
 
-        raw_defaults = dict(
-            name="ElectricalSeries",
-            description="Raw acquired data."
-        )
-        lfp_defaults = dict(
-            name="LFP",
-            description="Local field potential signal."
-        )
-        if metadata is None or not any([x in metadata['Ecephys'] for x in ['ElectricalSeries', 'LFPElectricalSeries']]):
-            metadata = dict(Ecephys=dict())
-            if not write_as_lfp:
-                metadata['Ecephys'].update(
-                    ElectricalSeries=dict(raw_defaults)
-                )
-            else:
-                metadata['Ecephys'].update(
-                    LFPElectricalSeries=dict(lfp_defaults)
-                )
-        assert ('ElectricalSeries' in metadata['Ecephys']
-                and isinstance(metadata['Ecephys']['ElectricalSeries'], dict)) \
-               or ('LFPElectricalSeries' in metadata['Ecephys']
-                   and isinstance(metadata['Ecephys']['LFPElectricalSeries'], dict)), \
-               ("Expected either metadata['Ecephys']['ElectricalSeries'] or "
-                "metadata['Ecephys']['LFPElectricalSeries'] to be a dictionary!")
-
-        if not write_as_lfp:
-            es_name = metadata['Ecephys']['ElectricalSeries'].get('name', raw_defaults['name'])
-            assert es_name not in nwbfile.acquisition, \
-                f"Raw ElectricalSeries '{es_name}' is already written in the NWBFile!"
-        else:
-            es_name = metadata['Ecephys']['LFPElectricalSeries'].get('name', lfp_defaults['name'])
+        assert write_as in ['raw', 'processed', 'lfp'], \
+            f"'write_as' should be 'raw', 'processed' or 'lfp', but intead received value {write_as}"
+        if write_as == 'raw':
+            eseries_kwargs = dict(
+                name="ElectricalSeries_raw",
+                description="Raw acquired data",
+                comments="Generated from SpikeInterface::NwbRecordingExtractor"
+            )
+        elif write_as == 'processed':
+            eseries_kwargs = dict(
+                name="ElectricalSeries_processed",
+                description="Processed data",
+                comments="Generated from SpikeInterface::NwbRecordingExtractor"
+            )
+        elif write_as == 'lfp':
+            eseries_kwargs = dict(
+                name="ElectricalSeries_lfp",
+                description="Processed data - LFP",
```

Comment:
Can you set this to its previous value?

Reply:
what previous value?

Reply:
https://github.com/SpikeInterface/spikeextractors/pull/621/files#diff-6dae9d8400e85905edd7719a4e2c19d9665c6a124cb08bda08cd09a15780ca85L699

Reply:
these default values shouldn't affect anything Also, they can be overwritten by user's metadata if wanted.